analyzers/conversion_overflow.go

```diff
@@ -161,6 +147,87 @@ func isStringToIntConversion(instr *ssa.Convert, dstType string) bool {
 	}
 }
 
+func hasExplicitRangeCheck(instr *ssa.Convert, dstType string) bool {
+	block := instr.Block()
+	dstInt, err := parseIntType(dstType)
+	if err != nil {
+		return false
+	}
+
+	minBoundChecked := false
+	maxBoundChecked := false
+
+	srcInt, err := parseIntType(instr.X.Type().String())
+	if err != nil {
+		return false
+	}
+
+	minBoundChecked = checkSourceMinBound(srcInt, dstInt)
+	maxBoundChecked = checkSourceMaxBound(srcInt, dstInt)
+
+	// If both bounds are already checked, return true
+	if minBoundChecked && maxBoundChecked {
+		return true
+	}
+
+	// Recursive function to check predecessors
+	var checkPredecessors func(block *ssa.BasicBlock) bool
+	checkPredecessors = func(block *ssa.BasicBlock) bool {
```

Comment:
```suggestion
	checkPredecessors := func(block *ssa.BasicBlock) bool {
```

Reply:
I can't do that since I call checkPredecessors inside itself so it needs to be predeclared in this case. Otherwise I get an undefined error